README.md

```diff
@@ -11,12 +11,17 @@ These interfaces can be used independently or together, depending on the use cas
 
 ## A note of caution
 
+This proposal contains two advanced features, `WeakRefs` and `FinalizationGroups`. Their correct use takes careful thought, and they are best avoided if possible.
+
 [Garbage collectors](https://en.wikipedia.org/wiki/Garbage_collection_(computer_science)) are complicated. If an application or library depends on GC cleaning up a WeakRef or calling a finalizer in a timely, predictable manner, it's likely to be disappointed: the cleanup may happen much later than expected, or not at all. Sources of variability include:
 - One object might be garbage-collected much sooner than another object, even if they become unreachable at the same time, e.g., due to generational collection.
 - Garbage collection work can be split up over time using incremental and concurrent techniques.
 - Various runtime heuristics can be used to balance memory usage, responsiveness.
 - The JavaScript engine may hold references to things which look like they are unreachable (e.g., in closures, or inline caches).
 - Different JavaScript engines may do these things differently, or the same engine may change its algorithms across versions.
+- Complex factors may lead to objects being held alive for unexpected amounts of time, such as use with certain APIs.
+
+If important logic is placed in the code path of a finalizer, it can transform memory management bugs into user-facing bugs. For example, if data is saved persistently from a finalizer, then a bug which accidentally keeps an additional reference around could lead to data loss.
```

Comment:
I think we might want to have this worded even more strongly. Since finalizers may never be executed simply because of different collector behaviors, important logic should NOT be using them.

> Important logic should not be placed in the code path of a finalizer. Doing so could create user-facing issues triggered by memory management bugs, or even differences between JavaScript garbage collector implementations. For example, if data is saved persistently solely from a finalizer, then a bug which accidentally keeps an additional reference around could lead to data loss.

Reply:
Adopted this wording; thanks!